### ISSUE TITLE
[v0.3.3 regression]: F811 incorrectly emitted if a symbol in a stub is re-exported and a symbol by the same name is defined in a different scope

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ruff#10509]
Starting in v0.3.3, Ruff incorrectly emits F811 on this snippet if it's a `.pyi` file:

```py
from foo import Bar as Bar

class Eggs:
    Bar: int  # F811 Redefinition of unused `Bar` from line 1
```

This shouldn't be flagged as a redefinition, as the `Eggs.Bar` variable is being defined in a different scope to the global `Bar` variable, which is being re-exported from the module. The regression bisects to 4b0666919 (by me).
